packages/api/endpoints/async-operations.js

```diff
@@ -42,7 +49,54 @@ async function getAsyncOperation(req, res) {
   return res.send(pick(asyncOperation, ['id', 'status', 'taskArn', 'description', 'operationType', 'output']));
 }
 
+/**
+ * Delete an async operation
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function deleteAsyncOperation(req, res) {
+  const {
+    asyncOperationModel = new AsyncOperationModel({
+      stackName: process.env.stackName,
+      systemBucket: process.env.system_bucket,
+      tableName: process.env.AsyncOperationsTable,
+    }),
+    asyncOperationPgModel = new AsyncOperationPgModel(),
+    knex = await getKnexClient(),
+  } = req.testContext || {};
+
+  const { id } = req.params;
+
+  let existingAsyncOperation;
+  try {
+    existingAsyncOperation = await asyncOperationModel.get({ id });
+  } catch (error) {
+    if (error instanceof RecordDoesNotExist) {
+      return res.boom.notFound('No record found');
+    }
+    throw error;
+  }
+
+  try {
+    await knex.transaction(async (trx) => {
+      await asyncOperationPgModel.delete(trx, { id });
+      await asyncOperationModel.delete({ id });
+    });
+    return res.send({ message: 'Record deleted' });
+  } catch (error) {
+    // Delete is idempotent, so there may not be a DynamoDB
+    // record to recreate
+    if (existingAsyncOperation) {
+      await asyncOperationModel.create(existingAsyncOperation);
```

Comment:
Will this mess with any timestamps on the record or anything like that? Are we sure it'll create _exactly_ the same record?

Reply:
Also, see this comment: https://github.com/nasa/cumulus/pull/2252/files#r638777192

Reply:
Resolved by removing the Dynamo create.